basetest.pm

```diff
@@ -715,6 +726,47 @@ sub parse_serial_output_qemu {
     return;
 }
 
+sub search_for_expected_autoinst_failures {
+    my ($self) = @_;
+    # autoinst failures defined in distri (test can override them)
+    my $failures = $self->{autoinst_failures};
+
+    my $die = 0;
+    my %regexp_matched;
+    # loop line by line
+    open(my $AUTOINSTLOG, "<", "autoinst-log.txt");
+    seek($AUTOINSTLOG, $autoinst_log_pos, 0);
+    while (my $line = <$AUTOINSTLOG>) {
+        chomp $line;
+        for my $regexp_table (@{$failures}) {
+            my $regexp  = $regexp_table->{pattern};
+            my $message = $regexp_table->{message};
+            my $type    = $regexp_table->{type};
+
+            # Input parameters validation
+            die "Wrong type defined for serial failure. Only 'soft' or 'hard' allowed. Got: $type" if $type !~ /^soft|hard|fatal$/;
```

Comment:
I don't care much but wouldn't it be better to do the validation already in `set_expected_autoinst_failures`?

Reply:
Leaving for now as discussed.